src/mscorlib/src/System/Globalization/EncodingTable.Unix.cs

```diff
@@ -1,101 +1,162 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Text;
+
 namespace System.Globalization
 {
-    using System;
-    using System.Text;
-    using System.Collections;
-    using System.Collections.Generic;
-    using System.Runtime.CompilerServices;
-    using System.Runtime.InteropServices;
-    using System.Runtime.Versioning;
-    using System.Security;
-    using System.Threading;
-    using System.Diagnostics.Contracts;
-
     internal static class EncodingTable
     {
+        //
+        // This caches the data item which is indexed by the code page value.
+        //
+        private static Hashtable hashByCodePage = Hashtable.Synchronized(new Hashtable());
+
         // Return a list of all EncodingInfo objects describing all of our encodings
         [System.Security.SecuritySafeCritical]  // auto-generated
-        internal static unsafe EncodingInfo[] GetEncodings()
+        internal static EncodingInfo[] GetEncodings()
         {
-            // TODO: Implement this fully.
-            return new EncodingInfo[] {
-                new EncodingInfo(CodePageUtf7, "utf-7", "Unicode (UTF-7)"),
-                new EncodingInfo(CodePageUtf8, "utf-8", "Unicode (UTF-8)"),
-                new EncodingInfo(CodePageUtf16, "utf-16", "Unicode"),
-                new EncodingInfo(CodePageUtf16BE, "utf-16BE", "Unicode (Big-Endian)"),
-                new EncodingInfo(CodePageUtf32, "utf-32", "Unicode (UTF-32)"),
-            };
-        }
-    
-        /*=================================GetCodePageFromName==========================
-        **Action: Given a encoding name, return the correct code page number for this encoding.
-        **Returns: The code page for the encoding.
-        **Arguments:
-        **  name    the name of the encoding
-        **Exceptions:
-        **  ArgumentNullException if name is null.
-        **  internalGetCodePageFromName will throw ArgumentException if name is not a valid encoding name.
-        ============================================================================*/
-        
-        internal static int GetCodePageFromName(String name)
-        {
-            // TODO: Implement this fully.
-            switch (name)
-            {
-                case "utf-7":
-                    return CodePageUtf7; 
+            EncodingInfo[] arrayEncodingInfo = new EncodingInfo[encodingDataTableItems.Length];
 
-                case "utf-8":
-                    return CodePageUtf8;
+            for (int i = 0; i < encodingDataTableItems.Length; i++)
+            {
+                CodePageDataItem dataItem = encodingDataTableItems[i];
 
-                case "utf-16":
-                    return CodePageUtf16;
+                arrayEncodingInfo[i] = new EncodingInfo(dataItem.CodePage, dataItem.WebName,
```

Comment:
I decided to keep it as is - creating the EncodingInfo each time the method is called - for a few reasons:
1. This is how the Windows code works.
2. The only caller of this method is the public Encoding.GetEncodings() method, which public callers can do whatever they want with the instances. So being consistent with Windows is good.
3. The public Encoding.GetEncodings() method isn't exposed in .NET Core, so pre-allocating and caching these EncodingInfos would be useless/wasteful on Unix.